src/node/txdownloadman_impl.cpp

```diff
@@ -142,4 +151,58 @@ void TxDownloadManagerImpl::DisconnectedPeer(NodeId nodeid)
     }
 
 }
+
+bool TxDownloadManagerImpl::AddTxAnnouncement(NodeId peer, const GenTxid& gtxid, std::chrono::microseconds now, bool p2p_inv)
+{
+    // If this is an inv received from a peer and we already have it, we can drop it.
+    if (p2p_inv && AlreadyHaveTx(gtxid, /*include_reconsiderable=*/true)) return true;
```

Comment:
1dd47af96cdc37706cd6d3f47892822e785b6fbe
What does the following mean: `p2p_inv=false` but `AlreadyHaveTx` is true?

Reply:
Grep will show `AddTxAnnouncement` called with `p2p_inv=false` in the new orphan handling code. We don't exit because we might already have a low feerate parent in reconsiderable rejects. You can also comment this out to see what fails.

Reply:
Agree the comment isn't 100% accurate, edited in #31190